### PR TITLE
feat(BreadcrumbLink): add BreadcrumbLink component

### PR DESCRIPTION
**Pull request by assuncaocharles**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

This PR adds `Breadcrumb.Link` component mainly to be used as anchors inside `Breadcrumb.Item` the main goal for it over using just `<a>` is to keep styles consistent:

With `<a>`:

Screen Shot 2020-09-19 at 14 26 08

While `Breadcrumb.Link`:

Screen Shot 2020-09-19 at 14 39 42

We are also adding `currrent` and `disabled` states together with and example showing sizes variations:

Screen Shot 2020-09-21 at 14 40 03

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/fluentui/react-northstar/src/components/Breadcrumb/BreadcrumbLink.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+} from '../../utils';
+import { useBreadcrumbContext } from './breadcrumbContext';
+import { Accessibility } from '@fluentui/accessibility';
+
+export interface BreadcrumbLinkProps
+  extends UIComponentProps<BreadcrumbLinkProps>,
+    ContentComponentProps,
+    ChildrenComponentProps {
+  /** Accessibility behavior if overridden by the user. */
+  accessibility?: Accessibility<never>;
```
Comment (pompomon) [inline packages/fluentui/react-northstar/src/components/Breadcrumb/BreadcrumbLink.tsx:27]: Please I think we would need accessibility for handling `disabled` props, WDYT?

Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Breadcrumb/BreadcrumbLink.tsx]: You are right, just did. :)

File: packages/fluentui/CHANGELOG.md [hunk 2]
```diff
@@ -60,6 +60,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Add arrow navigation for `Breadcrumb` a11y behavior @yuanboxue-amber ([#15087](https://github.com/microsoft/fluentui/pull/15087))
 - Updated `ReactionsIcon` and `MicrophoneDisabledIcon`. Added `CastingIcon`, `ComputerAudioIcon` and `ComputerAudioOffIcon`. @TanelVari ([#14998](https://github.com/microsoft/fluentui/pull/14998))
 - Added `CheckmarkCircleIcon`. Updated `ShareLocationIcon` @TanelVari ([#15119](https://github.com/microsoft/fluentui/pull/15119))
+- Added `BreadcrumbLink` component. @assuncaocharles ([#15136](https://github.com/microsoft/fluentui/pull/15136))
```
Comment (layershifter) [inline packages/fluentui/CHANGELOG.md:63]:
```suggestion
- Added `BreadcrumbLink` component @assuncaocharles ([#15136](https://github.com/microsoft/fluentui/pull/15136))
```

File: packages/fluentui/accessibility/src/behaviors/Breadcrumb/breadcrumbItemBehavior.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { Accessibility } from '../../types';
  * @specification
  * Adds role='listitem'.
  */
-export const breadcrumbItemBehavior: Accessibility<BreadcrumbItemBehaviorProps> = () => ({
+export const breadcrumbItemBehavior: Accessibility<BreadcrumbItemBehaviorProps> = props => ({
```
Comment (layershifter) [inline packages/fluentui/accessibility/src/behaviors/Breadcrumb/breadcrumbItemBehavior.ts:7]: Is it intentional?

File: packages/fluentui/docs/src/examples/components/Breadcrumb/Variations/BreadcrumbExampleSizes.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import { Breadcrumb, Flex, ChevronEndIcon, CallControlShareIcon } from '@fluentui/react-northstar';
+
+const BreadcrumbExampleSizes = props => (
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Breadcrumb/Variations/BreadcrumbExampleSizes.tsx:4]:
```suggestion
const BreadcrumbExampleSizes = () => (
```

File: packages/fluentui/docs/src/examples/components/Breadcrumb/Variations/index.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+import ComponentExample from '../../../../components/ComponentDoc/ComponentExample';
+import ExampleSection from '../../../../components/ComponentDoc/ExampleSection';
+
+const Variations = () => (
+  <ExampleSection title="Variations">
+    <ComponentExample
+      title="Default"
+      description="A default breadcrumb."
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Breadcrumb/Variations/index.tsx:9]: Please fix `title` and `description`

File: packages/fluentui/react-northstar/src/components/Breadcrumb/Breadcrumb.tsx [hunk 6]
```diff
@@ -30,9 +33,12 @@ export interface BreadcrumbProps
     ChildrenComponentProps {
   /** Accessibility behavior if overridden by the user. */
   accessibility?: Accessibility<BreadcrumbBehaviorProps>;
+
+  /** Breadcrumb can be sized */
+  size?: SizeValue;
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Breadcrumb/Breadcrumb.tsx:38]: Let's reduce this type to really available sizes

File: packages/fluentui/react-northstar/src/components/Breadcrumb/Breadcrumb.tsx [hunk 7]
```diff
@@ -82,7 +89,9 @@ export const Breadcrumb = compose<'nav', BreadcrumbProps, BreadcrumbStylesProps,
           ...unhandledProps,
         })}
       >
-        <div role="list">{childrenExist(children) ? children : content}</div>
+        <BreadcrumbContext.Provider value={{ size }}>
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Breadcrumb/Breadcrumb.tsx:92]: `value` should be memoized or `useContextSelector()` should be used

File: packages/fluentui/react-northstar/src/components/Breadcrumb/BreadcrumbItem.tsx [hunk 8]
```diff
@@ -9,18 +9,33 @@ import {
...
+import { useBreadcrumbContext } from './breadcrumbContext';
 
 export interface BreadcrumbItemProps
   extends UIComponentProps<BreadcrumbItemProps>,
     ContentComponentProps,
     ChildrenComponentProps {
   /** Accessibility behavior if overridden by the user. */
   accessibility?: Accessibility<BreadcrumbItemBehaviorProps>;
+
+  /** The Breadcrumb Link can be disabled */
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Breadcrumb/BreadcrumbItem.tsx:29]:
```suggestion
  /** The Breadcrumb Link can be disabled. */
```

File: packages/fluentui/react-northstar/src/components/Breadcrumb/BreadcrumbItem.tsx [hunk 9]
```diff
@@ -9,18 +9,33 @@ import {
...
+  disabled?: boolean;
+
+  /** Indicates if the link is the active */
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Breadcrumb/BreadcrumbItem.tsx:32]:
```suggestion
  /** Indicates if the link is the active. */
```

File: packages/fluentui/react-northstar/src/components/Breadcrumb/BreadcrumbLink.tsx [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+export type BreadcrumbLinkStylesProps = { size: SizeValue };
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Breadcrumb/BreadcrumbLink.tsx:31]: We should have this in props IMO

File: packages/fluentui/react-northstar/src/themes/teams/components/Breadcrumb/breadcrumbLinkStyles.ts [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+      display: 'flex',
+      alignItems: 'center',
+      position: 'relative',
+      ':visited': {
+        color: 'inherit',
+      },
+      ':focus': borderFocusStyles[':focus'],
+      ':focus-visible': {
+        ...borderFocusStyles[':focus-visible'],
+      },
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/themes/teams/components/Breadcrumb/breadcrumbLinkStyles.ts:54]:
```suggestion
      ...borderFocusStyles,
```

nit

File: packages/fluentui/react-northstar/src/themes/teams/components/Breadcrumb/breadcrumbLinkStyles.ts [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+
+      ...(p.size === 'smaller' && {
+        gap: v.linkSmallerGap,
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/themes/teams/components/Breadcrumb/breadcrumbLinkStyles.ts:57]: Our customer moves to Electron 8, it runs on Chromium 80, I would not be so optimistic about `gap` with Flexbox:

image

File: packages/fluentui/react-northstar/src/themes/teams/components/Breadcrumb/breadcrumbVariables.ts [hunk 13]
```diff
@@ -1,9 +1,58 @@
...
-export const breadcrumbVariables = (): BreadcrumbVariables => ({
+export const breadcrumbVariables = (siteVars): BreadcrumbVariables => ({
   dividerMargin: `0 ${pxToRem(8)}`,
+  fontSizeSmaller: pxToRem(12),
+  fontSizeSmall: pxToRem(12),
+  fontSizeMedium: pxToRem(14),
+  fontSizeLarge: pxToRem(18),
+
+  // Breadcrumb Item
+  currentItemFontWeight: siteVars.fontWeightBold,
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/themes/teams/components/Breadcrumb/breadcrumbVariables.ts:40]:
```suggestion
  itemCurrentFontWeight: siteVars.fontWeightBold,
```

File: packages/fluentui/react-northstar/src/themes/teams/components/Breadcrumb/breadcrumbLinkStyles.ts (hunk 12 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/themes/teams/components/Breadcrumb/breadcrumbLinkStyles.ts]: removed gap

File: packages/fluentui/react-northstar/src/components/Breadcrumb/BreadcrumbLink.tsx (hunk 10 above)
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Breadcrumb/BreadcrumbLink.tsx:31]: Oops, should exclude few values

File: packages/fluentui/react-northstar/src/components/Breadcrumb/BreadcrumbItem.tsx [hunk 14]
```diff
@@ -9,18 +9,33 @@ import {
...
+  /** The Breadcrumb Link can be disabled. */
+  disabled?: boolean;
+
+  /** Indicates if the link is the active. */
+  active?: boolean;
 }
 
-export type BreadcrumbItemStylesProps = never;
+export type BreadcrumbItemStylesProps = Required<Pick<BreadcrumbItemProps, 'active' | 'disabled'>> & {
+  size: SizeValue;
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Breadcrumb/BreadcrumbItem.tsx:37]: Same there

```suggestion
  size: Exclude<SizeValue, 'larger' | 'largest'>;
```

File: packages/fluentui/react-northstar/src/components/Breadcrumb/BreadcrumbLink.tsx (hunk 10 above)
Comment (layershifter) [reply packages/fluentui/react-northstar/src/components/Breadcrumb/BreadcrumbLink.tsx:31]:
```suggestion
export type BreadcrumbLinkStylesProps = { size: Exclude<SizeValue, 'larger' | 'largest'> };
```

File: packages/fluentui/react-northstar/src/themes/teams/getBorderFocusStyles.ts [hunk 15]
```diff
@@ -51,31 +32,42 @@ export const getBorderFocusStyles = (args: BorderFocusStyles): ICSSInJSStyle =>
...
+    borderRadius,
+  };
+  const borderPaddingTop = (borderPadding as BorderPadding)?.top || borderPadding;
+  const borderPaddingBottom = (borderPadding as BorderPadding)?.bottom || borderPadding;
+  const borderPaddingLeft = (borderPadding as BorderPadding)?.left || borderPadding;
+  const borderPaddingRight = (borderPadding as BorderPadding)?.right || borderPadding;
   return {
     ':focus': {
-      outline: 'none',
+      outline: 0,
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/themes/teams/getBorderFocusStyles.ts:49]:
```suggestion
      outline: 'none',
```

This is falsy change

File: packages/fluentui/react-northstar/src/components/Breadcrumb/breadcrumbContext.ts [hunk 16]
```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+import { SizeValue } from '../../utils';
+
+export type BreadcrumbContextValue = {
+  size: SizeValue;
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Breadcrumb/breadcrumbContext.ts:5]:
```suggestion
  size: Exclude<SizeValue, 'larger' | 'largest'>;
```

We can drop a type there and reuse in other components

File: packages/fluentui/react-northstar/src/components/Breadcrumb/breadcrumbContext.ts [hunk 17]
```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+import { BreadcrumbSizeValues } from './Breadcrumb';
+
+export type BreadcrumbContextValue = {
+  size: BreadcrumbSizeValues;
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Breadcrumb/breadcrumbContext.ts:5]: Let's have this type there to avoid circular imports

**Review by layershifter — APPROVED** [review]

**Pull request merged**